packages/atomic/src/components/commerce/atomic-commerce-products-per-page/atomic-commerce-products-per-page.tsx

```diff
@@ -0,0 +1,179 @@
+import {
+  Pagination,
+  PaginationState,
+  buildSearch,
+  buildProductListing,
+  SearchSummaryState,
+  ProductListingSummaryState,
+  Summary,
+} from '@coveo/headless/commerce';
+import {Component, Event, EventEmitter, h, Prop, State} from '@stencil/core';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {randomID} from '../../../utils/utils';
+import {FieldsetGroup} from '../../common/fieldset-group';
+import {PagerGuard} from '../../common/pager/pager-guard';
+import {RadioButton} from '../../common/radio-button';
+import type {CommerceBindings as Bindings} from '../atomic-commerce-interface/atomic-commerce-interface';
+
+/**
+ * The `atomic-commerce-products-per-page` component determines how many products to display per page.
+ *
+ * @part label - The "Products per page" label.
+ * @part buttons - The list of buttons.
+ * @part button - The product per page button.
+ * @part active-button - The active product per page button.
+ *
+ * @alpha
+ */
+@Component({
+  tag: 'atomic-commerce-products-per-page',
+  styleUrl: 'atomic-commerce-products-per-page.pcss',
+  shadow: true,
+})
+export class AtomicResultsPerPage implements InitializableComponent<Bindings> {
```

Comment:
There's too much copy paste between this and `atomic-results-per-page`. 

The most important thing will be to extract functional components for everything visual (ie: all `renderXYZ()` function) at the very least. This is mandatory to make sure parts, styling (css classes), i18n strings, and how things behave for end users stays consistent.

Then there's some other utility type function (ie: the `validateXYZ()` functions, or code that control end user focus for accessibility) can be shared exactly one to one, and so we should aim to refactor and extract those. 

In general however, those utility function are "less important" to refactor, if compared to the rendering code itself.

Reply:
I extracted the rendering logic in 2 seperate common component and added the validate logic to a common file.